src/isolate.rs

```diff
@@ -208,6 +183,8 @@ impl Isolate {
     let libdeno_isolate = unsafe { libdeno::deno_new(config) };
     // This channel handles sending async messages back to the runtime.
     let (tx, rx) = mpsc::channel::<(usize, Buf)>();
+    let isolate_perms =
+      Arc::new(permissions.unwrap_or(DenoPermissions::new(&state.flags)));
```

Comment:
I think this is strange that the permissions come from state.flags or from this extra argument. I'd rather explicitly have the caller create the permission struct and always pass it, than this implicit logic.